### ISSUE TITLE
The Vue app rebinds the widget at each change of a reactive property if the widget is configured with object property

### ISSUE THREAD

**Issue by veselints** [telerik/kendo-ui-core#3952]
### Bug report
The following initialization of the Grid:
```html
<kendo-grid id="projects_grid"
            :data-source-ref="'datasource1'"
            :resizable='true'
            :filterable='true'
            :sortable='true'
            :pageable='{ refresh: true, pageSize: 100, pageSizes: true, buttonCount: 5 }'>
```
will rebind the widget at each change of any reactive property present in the Vue app:
```html
<input v-model="new_proj_name" type="text" />
```
```JavaScript
export default {
  name: 'grid',
  data: function () {
    return {
      new_proj_name: ''
    }
  },
...
```

**Comment by veselints** [comment]
To workaround this issue use the kebab-case properties:
```html
<kendo-grid :dataSourceRef="'datasource1'"
            :resizable='true'
            :filterable='true'
            :sortable='true'
            :pageable-refresh='true'
            :pageable-page-size='100'
            :pageable-page-sizes='true'
            :pageable-button-count='5'
```

**Comment by ChuckFields** [comment]
I was having the same issue. I've implemented the Vue grid using websockets and was doing this:
`<kendo-grid :filterable='{mode: "row"}'`
This causes dataSource.transport.push() to be called once (like it normally does) and then once for every column that is of type: string. This causes the websocket event listener to get added multiple times.
This can be seen in the browser console [here](https://plnkr.co/edit/fMUcVGc3gGlQdqzWj43S?p=preview).
This was resolved using @veselints workaround by doing this:
`<kendo-grid :filterable-row='true'`

**Comment by umutozel** [comment]
Comparing JSON.stringify results is problematic, we had circular dependency problems with a few components.
Replacing that with a deep object comparing will solve the problem and it should work faster.

I couldn't find the vue wrapper repositories, I would be happy to fork and send PRs.

**Comment by Iankodj** [comment]
For the time being, the vue wrapper repository is private and cannot be forked. However, any ideas, suggestions and feedback are welcome and I encourage you to share them here or in a new GitHub issue.

**Comment by besmiralia** [comment]
What would be a temporary fix until the team fixes it permanently?

**Comment by Iankodj** [comment]
This particular issue is already fixed and live with the latest version of the Kendo UI for Vue components.
 
If you are referring to the other logged item, generally, I would recommend you to avoid using inline objects and arrays as values to the props. 

The main reason for the discussed behavior is that Vue dispatches a change for every inline object bound even if the change has nothing to do with the particular object. I believe, that is expected as observable properties cannot be applied to the anonymous inline objects bound thus they are bound on every data change there is. 

As for a workaround, one is shown in https://github.com/telerik/kendo-ui-core/issues/3952#issuecomment-361922607.  Another solution is to instead of using inline object, to have the object in the data field of the component. 

```html
<kendo-grid id="projects_grid"
            :data-source-ref="'datasource1'"
            :resizable='true'
            :filterable='true'
            :sortable='true'
            :pageable='pageable'>
...
```

```JavaScript
export default {
  name: 'grid',
  data: function () {
    return {
      new_proj_name: '',
      pageable: { refresh: true, pageSize: 100, pageSizes: true, buttonCount: 5 }
    }
  },
...
```

**Comment by umutozel** [comment]
kendo-base-component index.js line:59 causes "Converting circular structure to JSON" error.

if (JSON.stringify(oldValue) !== JSON.stringify(newValue))

We warned all development teams to avoid binding inline objects but some still use it.

So as an alternate solution, we created an new version of @progress/kendo-base-components-vue-wrapper package and published it to our proxy npm server, we changed those line with a object comparer + circular fixed json string comparer.

But you are right @Iankodj, best thing is the avoid using inline object bindings.
